modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/classloader/ArtifactClassLoaderConfigurationBuilder.java

```diff
@@ -0,0 +1,149 @@
+import org.mule.runtime.module.artifact.api.descriptor.BundleScope;
+import org.mule.runtime.module.artifact.api.descriptor.ClassLoaderModel;
+import org.mule.tools.api.classloader.model.AppClassLoaderModel;
+import org.mule.tools.api.classloader.model.Artifact;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+/**
+ * ClassLoaderModelBuilder that adds the concept of Shared Library for the configured dependencies.
+ */
+public class ArtifactClassLoaderConfigurationBuilder extends ClassLoaderModel.ClassLoaderModelBuilder {
+
+  private static final Logger LOGGER = getLogger(ArtifactClassLoaderConfigurationBuilder.class);
+
+  private final org.mule.tools.api.classloader.model.ClassLoaderModel packagerClassLoaderModel;
+  private final File artifactFolder;
+
+  public ArtifactClassLoaderConfigurationBuilder(org.mule.tools.api.classloader.model.ClassLoaderModel packagerClassLoaderModel,
+                                                 File artifactFolder) {
+    this.packagerClassLoaderModel = packagerClassLoaderModel;
+    this.artifactFolder = artifactFolder;
+  }
+
+  @Override
+  public ClassLoaderModel build() {
+    exportSharedLibrariesResourcesAndPackages();
+    processAdditionalPluginLibraries();
+
+    return super.build();
+  }
+
+  /**
+   * Exports shared libraries resources and packages getting the information from the packager
+   * {@link org.mule.tools.api.classloader.model.ClassLoaderModel}.
+   */
+  private void exportSharedLibrariesResourcesAndPackages() {
+    packagerClassLoaderModel.getDependencies().stream()
+        .filter(Artifact::isShared)
+        .filter(sharedDep -> !validateMuleRuntimeSharedLibrary(sharedDep.getArtifactCoordinates().getGroupId(),
+                                                               sharedDep.getArtifactCoordinates().getArtifactId()))
+        .forEach(sharedDep -> {
+          // if (isSupportingPackagesResourcesInformation()) {
+          this.exportingPackages(sharedDep.getPackages() == null ? emptySet() : new HashSet<>(asList(sharedDep.getPackages())));
+          this.exportingResources(sharedDep.getResources() == null ? emptySet()
+              : new HashSet<>(asList(sharedDep.getResources())));
+          // } else {
+          // findAndExportSharedLibrary(sharedDep.getArtifactCoordinates().getGroupId(),
+          // sharedDep.getArtifactCoordinates().getArtifactId());
+          // }
+        });
+  }
+
+  protected void processAdditionalPluginLibraries() {
+    if (packagerClassLoaderModel instanceof AppClassLoaderModel) {
+      AppClassLoaderModel appClassLoaderModel = (AppClassLoaderModel) packagerClassLoaderModel;
+      appClassLoaderModel.getAdditionalPluginDependencies()
+          .ifPresent(additionalDeps -> additionalDeps.forEach(this::updateDependency));
+    }
+  }
+
+  protected final boolean validateMuleRuntimeSharedLibrary(String groupId, String artifactId) {
```

Comment:
this is also in `AbstractArtifactClassLoaderConfigurationAssembler`, is this being validated more than once?

Reply:
Removed unnecessary validation in the builder.